src/components/periodic-report/periodic-report.edgedb.repository.ts

```diff
@@ -0,0 +1,189 @@
+import { Injectable } from '@nestjs/common';
+import { Query } from 'cypher-query-builder';
+import { Without } from 'type-fest/source/merge-exclusive';
+import {
+  CalendarDate,
+  ID,
+  PublicOf,
+  Range,
+  Session,
+  UnsecuredDto,
+} from '~/common';
+import { castToEnum, e, RepoFor } from '~/core/edgedb';
+import { Variable } from '../../core/database/query';
+import { ProgressReport } from '../progress-report/dto';
+import {
+  FinancialReport,
+  IPeriodicReport,
+  MergePeriodicReports,
+  NarrativeReport,
+  ReportType,
+  resolveReportType,
+} from './dto';
+import { PeriodicReportRepository } from './periodic-report.repository';
+
+@Injectable()
+export class PeriodicReportEdgeDBRepository
+  extends RepoFor(IPeriodicReport, {
+    hydrate: (periodicReport) => ({
+      ...periodicReport['*'],
+      type: castToEnum(periodicReport.__type__.name.slice(9, -7), ReportType),
+      reportFile: true,
+      sensitivity: periodicReport.container.is(e.Project.ContextAware)
+        .sensitivity,
+      scope: false,
+      parent: e.tuple({
+        identity: periodicReport.id,
+        labels: e.array_agg(e.set(periodicReport.__type__.name.slice(9, null))),
+        properties: e.tuple({
+          id: periodicReport.id,
+          createdAt: periodicReport.createdAt,
+        }),
+      }),
+    }),
+  })
+  implements PublicOf<PeriodicReportRepository>
+{
+  merge(
+    input: MergePeriodicReports,
+  ): Promise<ReadonlyArray<{ id: ID; interval: Range<CalendarDate> }>> {
+    throw new Error('Method not implemented.');
+  }
+
+  matchCurrentDue(
+    parentId: ID | Variable,
+    reportType: ReportType,
+  ): (query: Query) => Query {
+    throw new Error('Method not implemented.');
+  }
+
+  getByDate(
+    parentId: ID,
+    date: CalendarDate,
+    reportType: ReportType,
+    _session: Session,
+  ) {
+    const resource = e.cast(e.Resource, e.uuid(parentId));
+
+    const report = e.select(e.PeriodicReport, (report) => ({
+      filter: e.all(
+        e.set(
+          e.op(resource.id, '=', report.container.id),
+          e.op(report.start, '<=', date),
+          e.op(report.end, '>=', date),
+        ),
+      ),
+      ...report.is(resolveReportType(reportType)),
```

Comment:
@CarsonF - Could you weigh in here?  If I replace `resolveReportType(reportType)` with e.FinancialReport, for example, the typing is fine.  Something about my `resolveReportType` seems incorrect.  Any ideas?

Reply:
That `is()` doesn't do what you think it does. And I also cannot remember off the top of my head lol.

You could change which type you are selecting.
```ts
const res = EnhancedResource.of(resolveReportType(reportType))

e.select(res.db, (report) => ({
```

Reply:
This line:

```ts
const res = EnhancedResource.of(resolveReportType(reportType));
```

gives this error:

`Type 'string' is not assignable to type 'Pick<(Without<FinancialReport, (Without<NarrativeReport, ProgressReport> & ProgressReport) | (Without<...> & NarrativeReport)> & ((Without<...> & ProgressReport) | (Without<...> & NarrativeReport))) | (Without<...> & FinancialReport), "type">'.`

Do I need to change `resolveReportType` in some way?

Reply:
Well, I guess I could change the type of the `reportType` to `AnyReport`, but that then cascades to the Neo4j signature.  Which I might be able to solve by doing something like this:

```ts
node('node', `${resolveReportType(type).toString()}Report`),
```

over [here](https://github.com/SeedCompany/cord-api-v3/blob/develop/src/components/periodic-report/periodic-report.repository.ts#L308)...but that seems messy.

Reply:
```ts
const res = EnhancedResource.of(resolveReportType({ type: reportType }))
```
Also https://github.com/SeedCompany/cord-api-v3/blob/develop/src/components/periodic-report/dto/merge-periodic-reports.dto.ts#L29-L29

Reply:
Closer, but still this error on `type` in your snippet:

`Type 'Without<FinancialReport, (Without<NarrativeReport, ProgressReport> & ProgressReport) | (Without<ProgressReport, NarrativeReport> & NarrativeReport)> & Without<...> & ProgressReport' is not assignable to type '"Progress" | "Financial" | "Narrative"'.`

Reply:
The reportType is already the class not a string, so you don't need to resolve it.
```ts
const res = EnhancedResource.of(reportType)
```